CMakeLists.txt

```diff
@@ -113,14 +113,13 @@ if (MPI)
     file(GLOB parpacksrc_STAT_SRCS PARPACK/SRC/MPI/*.f)
 endif()
 
-#add_library(arpack SHARED
-#                   ${arpackblas_STAT_SRCS}
-#                   ${arpacklapack_STAT_SRCS}
-#                   ${arpackutil_STAT_SRCS}
-#                   ${arpacksrc_STAT_SRCS})
-add_library(arpack ${arpackutil_STAT_SRCS}
-                   ${arpacksrc_STAT_SRCS}
-                   ${arpacksrc_ICB})
+option(ENABLE_STATIC "Static linking (shared otherwise)" ON)
+if (ENABLE_STATIC)
+  add_library(arpack STATIC ${arpackutil_STAT_SRCS} ${arpacksrc_STAT_SRCS} ${arpacksrc_ICB})
```

Comment:
Can the STATIC/SHARED be moved into a variable to avoid duplicating the call to add_library() ?

Reply:
Not sure... I would say no (STATIC/SHARED are keywords). You may wrap that in a cmake function.